flink-runtime/src/main/java/org/apache/flink/runtime/state/filesystem/FsCheckpointStorageAccess.java

```diff
@@ -113,8 +113,14 @@ public boolean supportsHighlyAvailableStorage() {
 
     @Override
     public void initializeBaseLocationsForCheckpoint() throws IOException {
-        fileSystem.mkdirs(sharedStateDirectory);
-        fileSystem.mkdirs(taskOwnedStateDirectory);
+        if (!fileSystem.mkdirs(sharedStateDirectory)) {
```

Comment:
I'm wondering whether this check is sufficient. `mkdirs` javadoc says:
`@return true if at least one new directory has been created, false otherwise`

"at least one" confuses me; but probably an additional check here would be an overkill because it's only an optimization.

WDYT?